### PR TITLE
Add sequence number to blocks

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Required for equivocation and invalid block checks in the future. Also handy for the general clique oracle.

### Does this PR relate to an RChain JIRA issue? 
If applicable, add link to corresponding JIRA issue.

https://rchain.atlassian.net/browse/RHOL-421

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/BlockGenerator.scala [hunk 1]
```diff
@@ -25,8 +25,9 @@ trait BlockGenerator {
       deploys: Seq[Deploy] = Seq.empty[Deploy],
       tsHash: ByteString = ByteString.EMPTY): F[BlockMessage] =
     for {
-      chain  <- blockDagState[F].get
-      nextId = chain.currentId + 1
+      chain             <- blockDagState[F].get
+      nextId            = chain.currentId + 1
+      nextCreatorSeqNum = chain.currentSeqNum.getOrElse(creator, 0)
```
Comment (birchmd) [inline casper/src/test/scala/coop/rchain/casper/BlockGenerator.scala]: Why don't we need a `+ 1` here?

File: casper/src/main/scala/coop/rchain/casper/BlockDag.scala [hunk 2]
```diff
@@ -9,7 +9,8 @@ final case class BlockDag(idToBlocks: Map[Int, BlockMessage],
                           blockLookup: Map[BlockHash, BlockMessage],
                           childMap: Map[BlockHash, Set[BlockHash]],
                           latestMessages: Map[Validator, BlockHash],
-                          currentId: Int)
+                          currentId: Int,
+                          currentSeqNum: Map[Validator, Int])
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/BlockDag.scala]: So this `seqNum` is something like an Ethereum account nonce, but we only have them for validators.

Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/BlockDag.scala]: Yeah

File: casper/src/test/scala/coop/rchain/casper/BlockGenerator.scala (hunk 1 above)
Comment (KentShikama) [reply casper/src/test/scala/coop/rchain/casper/BlockGenerator.scala]: We totally do :)

Comment (KentShikama) [reply casper/src/test/scala/coop/rchain/casper/BlockGenerator.scala]: I wonder how it passed the tests...

Comment (birchmd) [reply casper/src/test/scala/coop/rchain/casper/BlockGenerator.scala]: That's what I was thinking...

**Comment by KentShikama** [comment]
@birchmd PTAL

**Pull request merged**